src/Compilers/CSharp/Test/Symbol/Symbols/ExtendedPartialMethodsTests.cs

```diff
@@ -0,0 +1,2018 @@
+
+partial class C
+{
+    internal virtual partial void M1([AllowNull] string s1);
+    internal virtual partial void M1(string s1) { }
+}
+class D : C
+{
+    internal override void M1(string s1) // 1
+    {
+    }
+}";
+            var comp = CreateCompilation(new[] { text1, AllowNullAttributeDefinition }, parseOptions: TestOptions.RegularWithExtendedPartialMethods);
+            comp.VerifyDiagnostics(
+                // (12,28): warning CS8765: Nullability of type of parameter 's1' doesn't match overridden member (possibly because of nullability attributes).
+                //     internal override void M1(string s1) // 1
+                Diagnostic(ErrorCode.WRN_TopLevelNullabilityMismatchInParameterTypeOnOverride, "M1").WithArguments("s1").WithLocation(12, 28)
+            );
+        }
+
+        [Fact]
+        public void Virtual_AllowNull_02()
+        {
+            const string text1 = @"
+#nullable enable
+using System.Diagnostics.CodeAnalysis;
+
+partial class C
+{
+    internal virtual partial void M1(string s1);
+    internal virtual partial void M1([AllowNull] string s1) { }
+}
+class D : C
+{
+    internal override void M1(string s1) // 1
+    {
+    }
+}";
+            var comp = CreateCompilation(new[] { text1, AllowNullAttributeDefinition }, parseOptions: TestOptions.RegularWithExtendedPartialMethods);
+            comp.VerifyDiagnostics(
+                // (12,28): warning CS8765: Nullability of type of parameter 's1' doesn't match overridden member (possibly because of nullability attributes).
+                //     internal override void M1(string s1) // 1
+                Diagnostic(ErrorCode.WRN_TopLevelNullabilityMismatchInParameterTypeOnOverride, "M1").WithArguments("s1").WithLocation(12, 28)
+            );
+        }
+
+        [Fact]
+        public void Virtual_AllowNull_03()
+        {
+            const string text1 = @"
+#nullable enable
+using System.Diagnostics.CodeAnalysis;
+
+partial class C
+{
+    internal virtual partial void M1(string s1);
+    internal virtual partial void M1([AllowNull] string s1) { }
+}
+class D : C
+{
+    internal override void M1([AllowNull] string s1)
+    {
+    }
+}";
+            var comp = CreateCompilation(new[] { text1, AllowNullAttributeDefinition }, parseOptions: TestOptions.RegularWithExtendedPartialMethods);
+            comp.VerifyDiagnostics();
+        }
+
+        [Fact]
+        public void Virtual_AllowNull_04()
+        {
+            const string text1 = @"
+#nullable enable
+using System.Diagnostics.CodeAnalysis;
+
+partial class C
+{
+    internal virtual partial void M1([AllowNull] string s1);
+    internal virtual partial void M1(string s1) { }
+}
+class D : C
+{
+    internal override void M1([AllowNull] string s1)
+    {
+    }
+}";
+            var comp = CreateCompilation(new[] { text1, AllowNullAttributeDefinition }, parseOptions: TestOptions.RegularWithExtendedPartialMethods);
+            comp.VerifyDiagnostics();
+        }
+
+        [Fact]
```

Comment:
Consider an `override` test where the types change from generic parameters to actual types. Example:

```cs
class Base<T> {
  protected virtual void M(T p) { } 
}

partial class Derived : Base<string> {
  protected override partial void M(string p); 
  protected override partial void M(string p) { }
}
```

Reply:
Adding to the test plan and will address in a follow-up PR.